code/main.lua

```diff
@@ -8,91 +8,195 @@ activity.setTitle('鲲')
 activity.setTheme(android.R.style.Theme_Material_Light)
 activity.setContentView(loadlayout("layout"))
 
-kun_is_back=0
-knowing = 0
-count2 = 0
-found_it3 = 0
-mode = 0
+WHAT_KUN_WANTS = 340
+ACHIEVEMENT_SUM = 7
+
 found = 0
-modechange = 0
-count = 0
-time=0
-ultimate=0
+mode = 0
+achievementCounter = 0
+
+notChanging = 0
+hittingModes = 0
+fortyTwo = 0
+cancelling = 0
+kunIsBack = 0
+calcAndCalc = 0
+knowingWhatKunWants = 0
+
+theEnd = 0
+
 
 calc.onClick=function(calculate)
-	time=time+1
-	if time==50 then
+	--the main part,because this is really a calculator
+	if mode==0 then
+		result.text=tostring(tonumber(adder1.text)+tonumber(adder2.text))
+	end
+	if mode==1 then
+		result.text=tostring(tonumber(adder1.text)-tonumber(adder2.text))
+	end
+	if mode==2 then
+		result.text=tostring(tonumber(adder1.text)*tonumber(adder2.text))
+	end
+	if mode==3 then
+		result.text=tostring(tonumber(adder1.text)/tonumber(adder2.text))
+	end
+
+
+	--Ifs
+
+	--0
+	if achievementCounter == 0 and mode == 3 and adder2.text == "0" and error_catch.checked then
+		achievementCounter = achievementCounter + 1
+	end
+	--1
+	if notChanging == 0 and mode == 0 and adder1.text == "20" and adder2.text == "10" then
+		notChanging = notChanging + 1
+	end
+	--2
+	if hittingModes > 0 then
+		hittingModes = 0
+	end
+	--3
+	if fortyTwo == 1 and mode == 0 and tonumber(adder1.text)+tonumber(adder2.text) == 42 then
+		fortyTwo = fortyTwo + 1
+	end
+	if fortyTwo == 1 and mode == 1 and tonumber(adder1.text)-tonumber(adder2.text) == 42 then
+		fortyTwo = fortyTwo + 1
+	end
+	if fortyTwo == 1 and mode == 2 and tonumber(adder1.text)*tonumber(adder2.text) == 42 then
+		fortyTwo = fortyTwo + 1
+	end
+	if fortyTwo == 1 and mode == 3 and tonumber(adder1.text)/tonumber(adder2.text) == 42 then
+		fortyTwo = fortyTwo + 1
+	end
+	--4
+	if cancelling > 0 then
+		cancelling = 0
+	end
+	--6
+	if calcAndCalc < 50 then 
```

Comment:
It just feels strange... I prefer to erase this `if` and make `calcAndCalc = calcAndCalc + 1` always executed. Only the `calcAndCalc == 50` will be granted the achievement right? I bet nobody could get it to inf...

Reply:
This is for formatting reasons.